src/main/java/com/commercetools/sync/commons/helpers/BaseSyncStatistics.java

```diff
@@ -0,0 +1,237 @@
+package com.commercetools.sync.commons.helpers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.netty.util.internal.StringUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.concurrent.TimeUnit;
+
+import static java.lang.String.format;
+
+public class BaseSyncStatistics {
+    private static final Logger LOGGER = LoggerFactory.getLogger(BaseSyncStatistics.class);
+
+    private String reportMessage;
+    private int updated;
+    private int created;
+    private int failed;
+    private int processed;
```

Comment:
I see that the only way to interact with this class is to create it and increment values. We cannot merge statistics prepared independently by two threads. Then the second approach with atomic counters is probably better.